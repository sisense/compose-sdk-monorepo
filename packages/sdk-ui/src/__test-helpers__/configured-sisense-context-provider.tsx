import { mockToken, mockUrl } from '@/__mocks__/msw';
import { SisenseContextProviderProps } from '@/props';
import { SisenseContextProvider } from '@/sisense-context/sisense-context-provider';

const contextProviderProps: SisenseContextProviderProps = {
  url: mockUrl,
  token: mockToken,
  defaultDataSource: 'Sample ECommerce',
  appConfig: {
    queryCacheConfig: { enabled: false },
    trackingConfig: { enabled: false },
  },
};

export const ConfiguredSisenseContextProvider: React.FC<{
  children: React.ReactNode;
}> = ({ children }) => {
  return <SisenseContextProvider {...contextProviderProps}>{children}</SisenseContextProvider>;
};
