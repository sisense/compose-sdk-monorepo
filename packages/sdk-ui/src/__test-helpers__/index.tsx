/**
 * Common test utilities
 */
import { PropsWithChildren, ReactElement } from 'react';
import { render, RenderResult } from '@testing-library/react';
import type { Cell, Data } from '@ethings-os/sdk-data';
import isObject from 'lodash-es/isObject';
import userEvent from '@testing-library/user-event';
import { SisenseContext, SisenseContextPayload } from '../sisense-context/sisense-context';
import { Authenticator, HttpClient } from '@ethings-os/sdk-rest-client';
import { ClientApplication } from '../app/client-application';
import { DeepPartial } from 'ts-essentials';
import { useTranslation } from 'react-i18next';
import { translation } from '../translation/resources/en';
import { CustomWidgetsProvider } from '../custom-widgets-provider';
import { MenuProvider } from '@/common/components/menu/menu-provider';

type UserSetupFn = (typeof userEvent)['setup'];
interface SetupResult extends RenderResult {
  user: ReturnType<UserSetupFn>;
}
/**
 * This helper method combines the userEvent.setup() and render() calls at the
 * start of a unit test using @testing-library/react.
 *
 * Based on recommendation by: https://testing-library.com/docs/user-event/intro/#writing-tests-with-userevent
 *
 * @param jsx - React component instance to test, passed directly to render()
 * @param useFakeTimers - true if Vitest fake timers are enabled
 * @returns
 */
export function setup(jsx: ReactElement, useFakeTimers = false): SetupResult {
  return {
    // Need to set "advanceTimers" if we're using Vitest's fake timers.
    // The solution in https://github.com/testing-library/user-event/issues/833#issuecomment-1171452841
    // is for Jest, but this seems to work for Vitest as well.
    user: userEvent.setup(
      useFakeTimers ? { advanceTimers: (ms) => vi.advanceTimersByTime(ms) } : undefined,
    ),
    ...render(jsx),
  };
}

export const MockedHighchartsReact = ({
  options,
}: {
  /**
   * Highcharts chart configuration object.
   * Please refer to the Highcharts (API documentation)[https://api.highcharts.com/highcharts/].
   */
  options: object;
}) => {
  return <div>{JSON.stringify(options)}</div>;
};

type MockedSisenseContextProviderProps = PropsWithChildren<
  DeepPartial<Omit<SisenseContextPayload, 'app'>>
>;

const mockHttpClient = new HttpClient(
  'http://mock-url/sometenant?someparam=true',
  {} as Authenticator,
  'mock-env',
);
export const MockedSisenseContextProvider = ({
  children,
  tracking = {
    enabled: true,
  },
  isInitialized = true,
  errorBoundary,
}: MockedSisenseContextProviderProps) => {
  return (
    <SisenseContext.Provider
      value={{
        app: {
          httpClient: mockHttpClient,
        } as ClientApplication,
        tracking: {
          enabled: tracking.enabled ?? true,
          packageName: tracking.packageName ?? 'sdk-ui',
        },
        isInitialized,
        errorBoundary: { showErrorBox: true, ...errorBoundary },
      }}
    >
      <MenuProvider>
        <CustomWidgetsProvider>{children}</CustomWidgetsProvider>
      </MenuProvider>
    </SisenseContext.Provider>
  );
};

/**
 * Extends the provided data by blurring specified rows.
 *
 * @param {Data} data - The original data object.
 * @param {number[]} blurRowsIndexes - An array of row indexes to be blurred.
 * @returns {Data} The data object with blurred rows.
 */
export const withBlurredRows = (data: Data, blurRowsIndexes: number[]) => ({
  ...data,
  rows: data.rows.map((row, rowIndex) =>
    row.map((value) => {
      const cell: Cell = isObject(value) ? value : { data: value };

      if (blurRowsIndexes.includes(rowIndex)) {
        return { ...cell, blur: true };
      }
      return cell;
    }),
  ),
});

/**
 * Mocking the i18next translation functionality for tests.
 */
export const setupI18nMock = () => {
  vi.mock('react-i18next', async () => {
    const original: { useTranslation: typeof useTranslation } = await vi.importActual(
      'react-i18next',
    );

    // Helper function to get nested translations safely
    const getNestedTranslation = (key: string): string | undefined => {
      return key.split('.').reduce((accumulator: any, currentKey: string) => {
        if (accumulator && typeof accumulator === 'object' && currentKey in accumulator) {
          return accumulator[currentKey];
        }
        return undefined;
      }, translation) as string | undefined;
    };

    return {
      ...original,
      useTranslation: () => ({
        ...original.useTranslation(),
        t: vi.fn((key, options?: Record<string, any>) => {
          const translatedValue = getNestedTranslation(key);
          if (translatedValue && options) {
            // Replace variables in the translated string
            return translatedValue.replace(/{{(.*?)}}/g, (_, varName) => {
              const trimmedVarName = varName.trim();
              return options[trimmedVarName] || `{{${trimmedVarName}}}`; // Fallback if variable is not found
            });
          }
          return translatedValue ?? key; // Fallback to the key if translation is not found
        }),
      }),
    };
  });
};
