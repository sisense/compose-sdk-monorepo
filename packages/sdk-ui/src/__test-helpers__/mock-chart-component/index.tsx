// Mocks the Chart component and provides a way to access the rendered charts.
// Use this to simulate user interactions with the chart while actual Chart component is mocked.

import { ChartProps } from '@/props';
import { RenderedChartModel } from './rendered-chart-model';

export type ChartMocksManager = {
  /**
   * Replace the actual Chart component with a mocked version.
   */
  mockChartComponent: () => void;
  /**
   * Restore the actual Chart component.
   */
  unmockChartComponent: () => void;
  /**
   * List of models of rendered charts.
   * Use these models to simulate user interactions with the rendered charts.
   */
  renderedCharts: RenderedChartModel[];
  /**
   * Clear all rendered charts from the mocks.
   */
  clearMocks: () => void;
};

const renderedCharts: RenderedChartModel[] = [];

const mockingConfig = {
  shouldChartBeMocked: false,
};

vi.mock('@/chart', async () => {
  const actualModule = await vi.importActual<typeof import('@/chart')>('@/chart');

  return {
    ...actualModule,
    Chart: (props: ChartProps) => {
      if (!mockingConfig.shouldChartBeMocked) {
        return actualModule.Chart(props); // Use the actual Chart component
      }
      const chartModel = new RenderedChartModel(
        props,
        `Chart_${renderedCharts.length} (${props.chartType})`,
      );
      renderedCharts.push(chartModel);
      return chartModel.render();
    },
  };
});

const mockChartComponent = () => {
  clearMocks();
  mockingConfig.shouldChartBeMocked = true;
};

const unmockChartComponent = () => {
  mockingConfig.shouldChartBeMocked = false;
};

const clearMocks = () => {
  renderedCharts.splice(0, renderedCharts.length);
};

export const chartMocksManager: ChartMocksManager = {
  mockChartComponent,
  unmockChartComponent,
  clearMocks,
  renderedCharts,
};
