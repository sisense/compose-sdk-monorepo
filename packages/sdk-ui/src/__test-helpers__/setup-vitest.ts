import matchers, { TestingLibraryMatchers } from '@testing-library/jest-dom/matchers';
import { cleanup } from '@testing-library/react';
import 'vitest-canvas-mock';

import createFetchMock from 'vitest-fetch-mock';
import { vi } from 'vitest';

// Add types for matchers from @testing-library/jest-dom
// https://github.com/testing-library/jest-dom/issues/439#issuecomment-1536524120
declare module 'vitest' {
  interface Assertion<T = any> extends jest.Matchers<void, T>, TestingLibraryMatchers<T, void> {}
}

// Manually extend Vitest's expect() with methods from @testing-library/jest-dom
expect.extend(matchers);

// Clean up mounted components after each test
// https://testing-library.com/docs/react-testing-library/api/#cleanup
afterEach(() => {
  cleanup();
});

vi.stubGlobal('__PACKAGE_VERSION__', '0.0.0');

// Mock ResizeObserver
const ResizeObserverMock = vi.fn(() => ({
  observe: vi.fn(),
  unobserve: vi.fn(),
  disconnect: vi.fn(),
}));

vi.stubGlobal('ResizeObserver', ResizeObserverMock);

// enable fetch mocking but don't mock anything by default
// and let tests decide to mock or not
const fetchMocker = createFetchMock(vi);
fetchMocker.enableMocks();
fetchMocker.dontMock();
