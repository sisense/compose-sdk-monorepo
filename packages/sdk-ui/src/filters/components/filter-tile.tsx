import type { FunctionComponent, ReactNode } from 'react';
import { useState } from 'react';

import { SisenseSwitchButton, TriangleIndicator } from './common';
import { ArrowDownIcon } from './icons';
import { useThemeContext } from '../../theme-provider';
import { getSlightlyDifferentColor } from '../../utils/color';
import { FilterVariant, isVertical } from './common/filter-utils';
import styled from '@emotion/styled';

interface Props {
  title: string;
  renderContent: (collapsed: boolean, tileDisabled: boolean) => ReactNode;
  arrangement?: FilterVariant;
  disabled?: boolean;
  isDependent?: boolean;
  onToggleDisabled?: () => void;
}

const GroupHoverWrapper = styled('div')({
  '.MuiSwitch-root': {
    opacity: 0.55,
    transition: 'all .3s ease',
  },
  '&:hover': {
    '.MuiSwitch-root': {
      opacity: 1,
    },
  },
});

/**
 * Generic component that owns common functionality of a filter "tile" like
 * collapsible content and an enable/disable toggle. This is intended to match
 * the style of filter tiles in the right sidebar on a Sisense dashboard.
 */
export const FilterTile: FunctionComponent<Props> = ({
  title,
  renderContent,
  arrangement = 'vertical',
  disabled,
  onToggleDisabled,
  isDependent,
}) => {
  const [collapsed, setCollapsed] = useState(true);

  const { themeSettings } = useThemeContext();

  const { backgroundColor: bgColor } = themeSettings.general;
  const { primaryTextColor: textColor } = themeSettings.typography;
  const disabledBgColor = getSlightlyDifferentColor(bgColor, 0.1);

  return (
    <GroupHoverWrapper>
      <div
        className={`${isVertical(arrangement) ? 'csdk-min-w-[216px]' : ''} ${
          isDependent ? '' : 'csdk-p-px'
        } csdk-w-min csdk-border csdk-border-solid csdk-border-[#dadada] csdk-text-text-content csdk-self-start`}
        style={{
          backgroundColor: disabled ? disabledBgColor : bgColor,
        }}
      >
        {isVertical(arrangement) && (
          <>
            {isDependent && <TriangleIndicator />}
            <header
              className={
                'csdk-flex csdk-items-center csdk-border-0 csdk-border-b csdk-border-solid csdk-border-b-[#dadada]'
              }
              style={{ color: textColor }}
            >
              <ArrowDownIcon
                aria-label="arrow-down"
                width="16"
                height="16"
                fill={`${textColor ?? '#5B6372'}`}
                className={`csdk-transition csdk-ml-[4px] csdk-cursor-pointer ${
                  collapsed ? '-csdk-rotate-90' : ''
                }`}
                onClick={() => setCollapsed((value) => !value)}
              />
              <span
                className={
                  'csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]'
                }
                style={{ color: textColor }}
              >
                {title}
              </span>
            </header>
          </>
        )}

        <main style={{ color: textColor }}>{renderContent(collapsed, disabled ?? false)}</main>

        {isVertical(arrangement) && (
          <footer
            className={
              'csdk-flex csdk-justify-end csdk-items-center csdk-min-h-[26px] csdk-border-0 csdk-border-t csdk-border-solid csdk-border-t-[#dadada]'
            }
          >
            <SisenseSwitchButton
              checked={!disabled}
              size="small"
              onChange={() => onToggleDisabled?.()}
              inputProps={{ role: 'switch', name: 'tile-switch' }}
            />
          </footer>
        )}
      </div>
    </GroupHoverWrapper>
  );
};
