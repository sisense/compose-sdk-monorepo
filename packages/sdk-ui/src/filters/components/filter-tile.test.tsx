/** @vitest-environment jsdom */
import { setup } from '@/__test-helpers__';
import { FilterTile } from './filter-tile';
import { screen } from '@testing-library/react';

describe('FilterTile', () => {
  it('should render delete button', async () => {
    setup(<FilterTile renderContent={() => <></>} onDelete={() => {}} />);
    const deleteButton = await screen.findByTestId('filter-delete-button');
    expect(deleteButton).toBeInTheDocument();
  });

  it('should not render delete button if onDelete is not provided', async () => {
    setup(<FilterTile renderContent={() => <></>} />);
    const deleteButton = screen.queryByTestId('filter-delete-button');
    expect(deleteButton).toBeNull();
  });

  it('should execute onDelete callback when delete button is clicked', async () => {
    const onDeleteMock = vi.fn();
    const { user } = setup(<FilterTile renderContent={() => <></>} onDelete={onDeleteMock} />);
    const deleteButton = await screen.findByTestId('filter-delete-button');
    await user.click(deleteButton);
    expect(onDeleteMock).toHaveBeenCalled();
  });
});
