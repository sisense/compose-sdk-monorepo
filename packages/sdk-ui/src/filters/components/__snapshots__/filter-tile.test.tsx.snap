// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterTile > renders CriteriaFilterTile for NumericFilter 1`] = `
<div>
  <div
    class="css-12vk4to"
  >
    <div
      class="csdk-accessible  css-l4aa93"
      style="min-width: 215px; background-color: rgb(255, 255, 255); font-family: "Open Sans","Roboto","Helvetica","Arial",sans-serif;"
    >
      <header
        class="css-3aw1ep"
        style="color: rgb(91, 99, 114);"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1lzjc8n-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-label="arrow-down"
            class="csdk-transition csdk-ml-[4px] csdk-cursor-pointer -csdk-rotate-90"
            data-testid="expand-collapse-button"
            fill="#5B6372"
            height="16"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.589 10L7 10.777 11.99 15 17 10.777 16.413 10l-4.422 3.727z"
              fill="#5B6372"
              fill-rule="nonzero"
            />
          </svg>
        </button>
        <span
          class="csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]"
          style="color: rgb(91, 99, 114); flex-grow: 1;"
        >
          Test Revenue
        </span>
      </header>
      <main
        style="color: rgb(91, 99, 114); position: relative;"
      >
        <div
          class="css-1uvk1au"
        >
          criteriaFilter.displayModePrefix criteriaFilter.greaterThan
        </div>
      </main>
      <footer
        class="css-1wpi5bx"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeSmall css-p0gqv1 css-1diaq70-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              name="tile-switch"
              role="switch"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </footer>
    </div>
  </div>
</div>
`;

exports[`FilterTile > renders CriteriaFilterTile for RankingFilter 1`] = `
<div>
  <div
    class="css-12vk4to"
  >
    <div
      class="csdk-accessible  css-l4aa93"
      style="min-width: 215px; background-color: rgb(255, 255, 255); font-family: "Open Sans","Roboto","Helvetica","Arial",sans-serif;"
    >
      <header
        class="css-3aw1ep"
        style="color: rgb(91, 99, 114);"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1lzjc8n-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-label="arrow-down"
            class="csdk-transition csdk-ml-[4px] csdk-cursor-pointer -csdk-rotate-90"
            data-testid="expand-collapse-button"
            fill="#5B6372"
            height="16"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.589 10L7 10.777 11.99 15 17 10.777 16.413 10l-4.422 3.727z"
              fill="#5B6372"
              fill-rule="nonzero"
            />
          </svg>
        </button>
        <span
          class="csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]"
          style="color: rgb(91, 99, 114); flex-grow: 1;"
        >
          Test Revenue
        </span>
      </header>
      <main
        style="color: rgb(91, 99, 114); position: relative;"
      >
        <div
          class="css-1uvk1au"
        >
          criteriaFilter.displayModePrefix criteriaFilter.top
        </div>
      </main>
      <footer
        class="css-1wpi5bx"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeSmall css-p0gqv1 css-1diaq70-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              name="tile-switch"
              role="switch"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </footer>
    </div>
  </div>
</div>
`;

exports[`FilterTile > renders CriteriaFilterTile for TextFilter 1`] = `
<div>
  <div
    class="css-12vk4to"
  >
    <div
      class="csdk-accessible  css-l4aa93"
      style="min-width: 215px; background-color: rgb(255, 255, 255); font-family: "Open Sans","Roboto","Helvetica","Arial",sans-serif;"
    >
      <header
        class="css-3aw1ep"
        style="color: rgb(91, 99, 114);"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1lzjc8n-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-label="arrow-down"
            class="csdk-transition csdk-ml-[4px] csdk-cursor-pointer -csdk-rotate-90"
            data-testid="expand-collapse-button"
            fill="#5B6372"
            height="16"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.589 10L7 10.777 11.99 15 17 10.777 16.413 10l-4.422 3.727z"
              fill="#5B6372"
              fill-rule="nonzero"
            />
          </svg>
        </button>
        <span
          class="csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]"
          style="color: rgb(91, 99, 114); flex-grow: 1;"
        >
          Test Revenue
        </span>
      </header>
      <main
        style="color: rgb(91, 99, 114); position: relative;"
      >
        <div
          class="css-1uvk1au"
        >
          criteriaFilter.displayModePrefix criteriaFilter.notStartsWith
        </div>
      </main>
      <footer
        class="css-1wpi5bx"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeSmall css-p0gqv1 css-1diaq70-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              name="tile-switch"
              role="switch"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </footer>
    </div>
  </div>
</div>
`;

exports[`FilterTile > renders CustomFilterTile for advnced filter type 1`] = `
<div>
  <div
    class="css-12vk4to"
  >
    <div
      class="csdk-accessible  css-l4aa93"
      style="min-width: 215px; background-color: rgb(255, 255, 255); font-family: "Open Sans","Roboto","Helvetica","Arial",sans-serif;"
    >
      <header
        class="css-3aw1ep"
        style="color: rgb(91, 99, 114);"
      >
        <span
          class="csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]"
          style="color: rgb(91, 99, 114); flex-grow: 1;"
        >
          Test Revenue
        </span>
      </header>
      <main
        style="color: rgb(91, 99, 114); position: relative;"
      >
        <div
          class="css-1uvk1au"
        >
          customFilterTileMessage
        </div>
      </main>
      <footer
        class="css-1wpi5bx"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeSmall css-p0gqv1 css-1diaq70-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              name="tile-switch"
              role="switch"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </footer>
    </div>
  </div>
</div>
`;

exports[`FilterTile > renders Filter Tile for unsupported filter type 1`] = `
<div>
  <div
    class="css-12vk4to"
  >
    <div
      class="csdk-accessible  css-l4aa93"
      style="min-width: 215px; background-color: rgb(255, 255, 255); font-family: "Open Sans","Roboto","Helvetica","Arial",sans-serif;"
    >
      <header
        class="css-3aw1ep"
        style="color: rgb(91, 99, 114);"
      >
        <span
          class="csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]"
          style="color: rgb(91, 99, 114); flex-grow: 1;"
        >
          Unsupported Filter
        </span>
      </header>
      <main
        style="color: rgb(91, 99, 114); position: relative;"
      >
        <div
          class="css-1uvk1au"
        >
          unsupportedFilterMessage
        </div>
      </main>
      <footer
        class="css-1wpi5bx"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeSmall css-p0gqv1 css-1diaq70-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              name="tile-switch"
              role="switch"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </footer>
    </div>
  </div>
</div>
`;

exports[`FilterTile > renders RelativeDateFilterTile for RelativeDateFilter 1`] = `
<div>
  <div
    class="css-12vk4to"
  >
    <div
      class="csdk-accessible  css-l4aa93"
      style="min-width: 215px; background-color: rgb(255, 255, 255); font-family: "Open Sans","Roboto","Helvetica","Arial",sans-serif;"
    >
      <header
        class="css-3aw1ep"
        style="color: rgb(91, 99, 114);"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1lzjc8n-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-label="arrow-down"
            class="csdk-transition csdk-ml-[4px] csdk-cursor-pointer -csdk-rotate-90"
            data-testid="expand-collapse-button"
            fill="#5B6372"
            height="16"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.589 10L7 10.777 11.99 15 17 10.777 16.413 10l-4.422 3.727z"
              fill="#5B6372"
              fill-rule="nonzero"
            />
          </svg>
        </button>
        <span
          class="csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]"
          style="color: rgb(91, 99, 114); flex-grow: 1;"
        >
          Years
        </span>
      </header>
      <main
        style="color: rgb(91, 99, 114); position: relative;"
      >
        <div
          class="css-1uvk1au"
        >
          dateFilter.next 1 dateFilter.years dateFilter.from dateFilter.today
        </div>
      </main>
      <footer
        class="css-1wpi5bx"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeSmall css-p0gqv1 css-1diaq70-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              name="tile-switch"
              role="switch"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </footer>
    </div>
  </div>
</div>
`;

exports[`FilterTile > renders locked CriteriaFilterTile 1`] = `
<div>
  <div
    class="css-12vk4to"
  >
    <div
      class="csdk-accessible  css-l4aa93"
      style="min-width: 215px; background-color: rgb(255, 255, 255); font-family: "Open Sans","Roboto","Helvetica","Arial",sans-serif;"
    >
      <header
        class="css-3aw1ep"
        style="color: rgb(91, 99, 114);"
      >
        <svg
          fill="#b4b4b4"
          height="22"
          viewBox="0 0 22 22"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(5.000000,5.000000) scale(.70, 0.6)"
          >
            <path
              d="m3,9v11h14V9M4,9V6c0-3.3 2.7-6 6-6c3.3,0 6,2.7 6,6v3H14V6c0-2.2-1.8-4-4-4-2.2,0-4,1.8-4,4v3"
            />
            <rect
              fill="#dcdcdc"
              height="9"
              width="12"
              x="4"
              y="10"
            />
          </g>
        </svg>
        <span
          class="csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]"
          style="color: rgb(91, 99, 114); flex-grow: 1;"
        >
          Test Revenue
        </span>
      </header>
      <main
        style="color: rgb(91, 99, 114); position: relative;"
      >
        <div
          class="css-1uvk1au"
        >
          criteriaFilter.displayModePrefix criteriaFilter.greaterThan
        </div>
        <div
          style="position: absolute; top: 4px; left: 4px; right: 4px; bottom: 4px; background-color: white; opacity: 0.5;"
        />
      </main>
      <footer
        class="css-1wpi5bx"
      />
    </div>
  </div>
</div>
`;

exports[`FilterTile > renders locked RelativeDateFilter component 1`] = `
<div>
  <div
    class="css-12vk4to"
  >
    <div
      class="csdk-accessible  css-l4aa93"
      style="min-width: 215px; background-color: rgb(255, 255, 255); font-family: "Open Sans","Roboto","Helvetica","Arial",sans-serif;"
    >
      <header
        class="css-3aw1ep"
        style="color: rgb(91, 99, 114);"
      >
        <svg
          fill="#b4b4b4"
          height="22"
          viewBox="0 0 22 22"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(5.000000,5.000000) scale(.70, 0.6)"
          >
            <path
              d="m3,9v11h14V9M4,9V6c0-3.3 2.7-6 6-6c3.3,0 6,2.7 6,6v3H14V6c0-2.2-1.8-4-4-4-2.2,0-4,1.8-4,4v3"
            />
            <rect
              fill="#dcdcdc"
              height="9"
              width="12"
              x="4"
              y="10"
            />
          </g>
        </svg>
        <span
          class="csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]"
          style="color: rgb(91, 99, 114); flex-grow: 1;"
        >
          Years
        </span>
      </header>
      <main
        style="color: rgb(91, 99, 114); position: relative;"
      >
        <div
          class="css-1uvk1au"
        >
          dateFilter.next 1 dateFilter.years dateFilter.from dateFilter.today
        </div>
        <div
          style="position: absolute; top: 4px; left: 4px; right: 4px; bottom: 4px; background-color: white; opacity: 0.5;"
        />
      </main>
      <footer
        class="css-1wpi5bx"
      />
    </div>
  </div>
</div>
`;
